api/handlers/listhandler.py

```
@@ -471,10 +477,16 @@ def get(self, cont_name, list_name, **kwargs):
             # START of duplicated code
             # IMPORTANT: If you modify the below code reflect the code changes in
             # refererhandler.py:AnalysesHandler's download method
-            signed_url = files.get_signed_url(file_path, file_system,
+            signed_url = None
+            if config.storage.is_signed_url():
+                try:
+                    signed_url = config.storage.get_signed_url(fileinfo.get('_id'), file_path,
                                               filename=filename,
                                               attachment=(not self.is_true('view')),
                                               response_type=str(fileinfo.get('mimetype', 'application/octet-stream')))
+                except fs.errors.ResourceNotFound:
+                    self.log.error('Error getting signed_url on non existing file')
```

Do we want to just log and error and move on in this instance? It's not clear from a high-level view what it would move on to in this case, but I might just be misreading.

---

I am not certain on this either.  I assume it because if you are trying to get parts of the file you wont have a signed url for the individual parts so then want to proceed through to the rest of the query params for Range and what not

---

Looking at the structure of the code, we should not bury the exception here, if we get a ResourceNotFound, I believe we should be raising an `APINotFoundException`.

---

This actually is never reached in testing - we bomb out on [line 440](https://github.com/flywheel-io/core/pull/1532/files#diff-2b6de8dffa100181ea2c9b85e6e5f50cR440) if the file doesn't exist.

I would still argue that we should just go ahead and raise an `APINotFoundException`

---

I found the specific case where returning a APINotFoundException is not working.  It happens when we have a storage provider that uses signedUrls but there are files in the storage provider using the legacy CAS file names.  The issue is the signed url can not be found since it assumes its using the uuid based filepaths and not the legacy CAS file paths.  It then fails over and just serves the file directly rather than using a signed URL.  If we can assure that no legacy CAS file structure is used on production instances then we can return the APINotFound exception.  But there is a specific integration test that checks this in test_downloads.py so I assume there must have been some customers where this check was needed.   I will leave the code as is for now.